From 6e83bc4b4a1e87ace5ab808a0b5a28cbf2b0821c Mon Sep 17 00:00:00 2001
From: Julien Gilli <julien.gilli@joyent.com>
Date: Thu, 28 Sep 2017 16:11:22 -0700
Subject: [PATCH] DAPI-345 filter out CNs not matching platform requirements
 for VMs that mount volumes Reviewed by: Marsell Kukuljevic
 <marsell@joyent.com> Approved by: Orlando Vazquez <orlando@joyent.com>

---
 lib/designation.js            | 15 +++++++++++++--
 package.json                  |  2 +-
 sapi_manifests/cnapi/template |  1 +
 tools/rsync-to                |  1 -
 4 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/lib/designation.js b/lib/designation.js
index 3a94f8d..3f7c58a 100644
--- a/lib/designation.js
+++ b/lib/designation.js
@@ -30,7 +30,15 @@ var DEFAULT_FILTER_HEADNODE      = true;
 var DEFAULT_FILTER_MIN_RESOURCES = true;
 var DEFAULT_FILTER_LARGE_SERVERS = true;
 var DEFAULT_DISABLE_OVERRIDE_OVERPROV = false;
-
+/*
+ * Duplicated from sdcadm, because we don't want to provide a SAPI configuration
+ * knob to users/operators that would potentially have them use a minimum
+ * platform older than our recommended minimum platform.
+ */
+var DEFAULT_FILTER_DOCKER_NFS_VOLUMES_AUTOMOUNT_MIN_PLATFORM =
+    '20160613T123039Z';
+var DEFAULT_FILTER_NON_DOCKER_NFS_VOLUMES_AUTOMOUNT_MIN_PLATFORM =
+    '20170925T211846Z';
 
 function Designation() {}
 
@@ -145,7 +153,10 @@ function getDefaults(changeDefaults) {
     setDefault('filter_headnode',      DEFAULT_FILTER_HEADNODE);
     setDefault('filter_min_resources', DEFAULT_FILTER_MIN_RESOURCES);
     setDefault('filter_large_servers', DEFAULT_FILTER_LARGE_SERVERS);
-
+    setDefault('filter_docker_nfs_volumes_automount_min_platform',
+        DEFAULT_FILTER_DOCKER_NFS_VOLUMES_AUTOMOUNT_MIN_PLATFORM);
+    setDefault('filter_non_docker_nfs_volumes_automount_min_platform',
+        DEFAULT_FILTER_NON_DOCKER_NFS_VOLUMES_AUTOMOUNT_MIN_PLATFORM);
     setDefault('weight_current_platform', DEFAULT_WEIGHT_CURRENT_PLATFORM);
     setDefault('weight_next_reboot',      DEFAULT_WEIGHT_NEXT_REBOOT);
     setDefault('weight_num_owner_zones',  DEFAULT_WEIGHT_NUM_OWNER_ZONES);
diff --git a/package.json b/package.json
index 3285474..f6d23c0 100644
--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
     "assert-plus": "1.0.0",
     "async": "1.5.2",
     "bunyan": "1.8.5",
-    "dapi": "git+https://github.com/joyent/sdc-designation.git#fb53f9d5",
+    "dapi": "git+https://github.com/joyent/sdc-designation.git#1fe6da24fdf1c948713e1857cf6b87e11c021788",
     "deep-equal": "1.0.1",
     "dox": "0.9.0",
     "effluent-logger": "git+https://github.com/joshwilsdon/effluent-logger.git#d662f161a07f94045ad2afb45442931511c40e51",
diff --git a/sapi_manifests/cnapi/template b/sapi_manifests/cnapi/template
index 283f667..5af56aa 100644
--- a/sapi_manifests/cnapi/template
+++ b/sapi_manifests/cnapi/template
@@ -80,6 +80,7 @@
 			        "hard-filter-min-free-disk",
 			        "hard-filter-vlans",
 			        "hard-filter-platform-versions",
+			        "hard-filter-feature-min-platform",
 			        "hard-filter-traits",
 			        "hard-filter-headnode",
 			        "override-overprovisioning",
diff --git a/tools/rsync-to b/tools/rsync-to
index f17bb3e..2fb85a2 100755
--- a/tools/rsync-to
+++ b/tools/rsync-to
@@ -42,7 +42,6 @@ rsync -av ${TOP}/ \
     --exclude /config/config.json \
     --exclude /deps/ \
     --exclude /docs/ \
-    --exclude /node_modules/ \
     --exclude /scripts/ \
     --exclude /boot/ \
     --exclude /smf/ \
-- 
2.21.0

